File: 프로그래머스/unrated/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.cs
using System;

public class Solution {
    public int solution(int[,] board, int k) 
    {
        int answer = 0;
        
        for(int y = 0; y < board.GetLength(0); ++y)
        {
            for(int x = 0; x < board.GetLength(1); ++x)
            {
                if(y + x <= k)
                    answer += board[y, x];
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, int k) {
    int answer = 0;
    
    for (int y = 0; y <= k && y < board.size(); ++y)
    {       
        for (int x = 0; x <= k && x < board[y].size(); ++x)
        {
            if (y + x > k)
                break;
            
            answer += board[y][x];
        }
    }
    
    return answer;
}